AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -1858,25 +1874,25 @@ - (ASLayout *)calculateLayoutThatFits:(ASSizeRange)constrainedSize
 
 - (CGSize)calculateSizeThatFits:(CGSize)constrainedSize
 {
-  ASDisplayNodeAssertThreadAffinity(self);
+  ASDN::MutexLocker l(_propertyLock);
   return _preferredFrameSize;
 }
 
 - (ASLayoutSpec *)layoutSpecThatFits:(ASSizeRange)constrainedSize
 {
-  ASDisplayNodeAssertThreadAffinity(self);
+  ASDN::MutexLocker l(_propertyLock);
   return nil;
 }
 
 - (ASLayout *)calculatedLayout
 {
-  ASDisplayNodeAssertThreadAffinity(self);
+  ASDN::MutexLocker l(_propertyLock);
   return _layout;
 }
 
 - (CGSize)calculatedSize
 {
-  ASDisplayNodeAssertThreadAffinity(self);
+  ASDN::MutexLocker l(_propertyLock);
```

Comment:
Add a property lock call to the method below this, -constrainedSizeForCalculatedLayout